ash/src/vk/prelude.rs

```diff
@@ -56,3 +56,7 @@ impl From<vk::Extent2D> for vk::Rect2D {
         }
     }
 }
+
+pub trait HasStructureType {
```

Comment:
bikeshed: `TypedStructure`? `TaggedStructure`? `Tagged`?

Reply:
I think I only mentioned it on the macro but I was very much looking for naming suggestions here too. `TaggedStructure` seems the most fitting to me as it's already "typed" in the Rust sense. Applied all your suggestions, thanks :+1: 

Then, do you want to take this trait one step further?

- (default) implementation `fn in()` and `fn out()` to cast to the respective `Base*Structure`;
- Reuse that in `push_next()`, either by adding a trait bound on the function or require `TaggedStructure` for every `Extends*` trait;
- Helpers to get/find structures of a specific `s_type` in the chain, like #628?

Reply:
> (default) implementation fn in() and fn out() to cast to the respective Base*Structure;

I'm on board in theory, but needs more design. Should these return pointers or references? Mutable or shared, or double the methods for both?

> Helpers to get/find structures of a specific s_type in the chain

I don't feel like I understand the use case for that (or the macro this PR introduces, for that matter) well enough to judge usefulness. Are we trying to support people implementing drivers/layers?

Reply:
Good question, since we have `push_next()` hiding this detail it is perhaps less relevant than I originally thought, to waste compiletime over.

Same for the getter functions, those would be implemented using `match_*_struct` _anyway_ and the user has to deal with the possibility of a certain typed struct appearing in the chain more than once. Putting it on the user to implement this with a `for` loop at least requires them to think about the possibility.

Reply:
:+1:, easier to add stuff in the future if use cases arise.

Reply:
Only the trait bound (`Extends*` requiring this) may be interesting to consider up front, but no use yet so no need to add it I guess.

Reply:
Yeah, clever trait hierarchies that we don't actually need for anything sounds like a trap.

Reply:
I don't consider it particularly clever or special, just something that exists. Users needing it outside of `ash` can simply expand their trait bounds if they want/need.